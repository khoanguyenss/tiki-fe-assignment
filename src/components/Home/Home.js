import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => (
  <div>
    <Link to="/beginner">
      <div>
        Beginner
      </div>
    </Link>
    <Link to="/advantage">
      <div>
        Advantage
      </div>
    </Link>
  </div>
);

export default Home;
