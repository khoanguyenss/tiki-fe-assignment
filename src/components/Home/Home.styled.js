import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const LevelBtn = styled(Link)`
  color: #000;
  margin-bottom: 25px;
  text-decoration: none;
  font-size: 24px;
  
  &:hover {
    text-decoration: underline;
  }
`;
