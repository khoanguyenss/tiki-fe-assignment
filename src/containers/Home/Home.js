import React from 'react';
import Home from '../../components/Home';
import { actions as MineSweeperActions } from '../../reducers/minesweeper';
import { useActions } from '../../utils/hooks';

const HomeContainer = () => {
  const resetBoard = useActions(
    MineSweeperActions.resetBoard,
  );

  return (
    <Home resetBoard={resetBoard} />
  );
};

export default HomeContainer;
