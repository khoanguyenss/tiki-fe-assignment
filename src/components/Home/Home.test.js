import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import Home from './Home';


describe('<Home />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  const resetBoardMock = jest.fn();

  const Component = mount(
    <Router>
      <Home
        resetBoard={resetBoardMock}
      />
    </Router>
    ,
  );

  it('should render', () => {
    expect(Component.length).toBe(1);
  });
  it('match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
  it('should render 2 Link', () => {
    const LinkComp = Component.find(Link);
    expect(LinkComp.length).toBe(2);
  });
  it('should call fetchBoardMock once', () => {
    // trigger useEffect
    act(() => {
      mount(
        <Router>
          <Home
            resetBoard={resetBoardMock}
          />
        </Router>
        ,
      );
    });
    expect(resetBoardMock).toBeCalledTimes(1);
  });
});
