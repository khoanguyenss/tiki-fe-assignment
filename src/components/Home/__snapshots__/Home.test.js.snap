// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home
      resetBoard={[MockFunction]}
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "bEeQEn",
                "rules": Array [
                  "
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-bdVaJa bEeQEn"
          >
            <Styled(Link)
              to="/beginner"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "gHlQFz",
                      "rules": Array [
                        "
  color: #000;
  margin-bottom: 25px;
  text-decoration: none;
  font-size: 24px;
  
  &:hover {
    text-decoration: underline;
  }
",
                      ],
                    },
                    "displayName": "Styled(Link)",
                    "foldedComponentIds": Array [],
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "Link",
                      "propTypes": Object {
                        "innerRef": [Function],
                        "onClick": [Function],
                        "replace": [Function],
                        "target": [Function],
                        "to": [Function],
                      },
                      "render": [Function],
                    },
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                to="/beginner"
              >
                <Link
                  className="sc-bwzfXH gHlQFz"
                  to="/beginner"
                >
                  <LinkAnchor
                    className="sc-bwzfXH gHlQFz"
                    href="/beginner"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-bwzfXH gHlQFz"
                      href="/beginner"
                      onClick={[Function]}
                    >
                      Beginner
                    </a>
                  </LinkAnchor>
                </Link>
              </StyledComponent>
            </Styled(Link)>
            <Styled(Link)
              to="/advantage"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "gHlQFz",
                      "rules": Array [
                        "
  color: #000;
  margin-bottom: 25px;
  text-decoration: none;
  font-size: 24px;
  
  &:hover {
    text-decoration: underline;
  }
",
                      ],
                    },
                    "displayName": "Styled(Link)",
                    "foldedComponentIds": Array [],
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "Link",
                      "propTypes": Object {
                        "innerRef": [Function],
                        "onClick": [Function],
                        "replace": [Function],
                        "target": [Function],
                        "to": [Function],
                      },
                      "render": [Function],
                    },
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                to="/advantage"
              >
                <Link
                  className="sc-bwzfXH gHlQFz"
                  to="/advantage"
                >
                  <LinkAnchor
                    className="sc-bwzfXH gHlQFz"
                    href="/advantage"
                    navigate={[Function]}
                  >
                    <a
                      className="sc-bwzfXH gHlQFz"
                      href="/advantage"
                      onClick={[Function]}
                    >
                      Advantage
                    </a>
                  </LinkAnchor>
                </Link>
              </StyledComponent>
            </Styled(Link)>
          </div>
        </StyledComponent>
      </styled.div>
    </Home>
  </Router>
</BrowserRouter>
`;
